sql-prepare-plan-cache.md

```diff
@@ -14,41 +14,47 @@ The TiDB optimizer handles these two types of queries in the same way: when prep
 
 When the execution plan cache is enabled, in the first execution every `Prepare` statement checks whether the current query can use the execution plan cache, and if the query can use it, then put the generated execution plan into a cache implemented by LRU (Least Recently Used) linked list. In the subsequent `Execute` queries, the execution plan is obtained from the cache and checked for availability. If the check succeeds, the step of generating an execution plan is skipped. Otherwise, the execution plan is regenerated and saved in the cache.
 
-In the current version of TiDB, when the `Prepare` statement meets any of the following conditions, the query cannot use the execution plan cache:
-
-- The query contains variables other than `?` (including system variables or user-defined variables);
-- The query contains sub-queries;
-- The query contains functions that cannot be cached, such as `current_user()`, `database()`, and `last_insert_id()`;
-- The `Order By` statement of the query contains `?`;
-- The `Group By` statement of the query contains `?`;
-- The `Limit [Offset]` statement of the query contains `?`;
-- The window frame definition of the `Window` function contains `?`;
-- Partition tables are involved in the query.
+In the current version of TiDB, if a `Prepare` statement meets any of the following conditions, the query or the plan is not cached:
+
+- The query contains SQL statements other than `SELECT`, `UPDATE`, `INSERT`, `DELETE`, `Union`, `Intersect`, and `Except`.
+- The query accesses partitioned tables or temporary tables, or a table that contains generated columns.
+- The query contains sub-queries, such as `select * from t where a > (select ...)`.
+- The query contains the `ignore_plan_cache` hint, such as `select /*+ ignore_plan_cache() */ * from t`.
+- The query contains variables other than `?` (including system variables or user-defined variables), such as `select * from t where a>? and b>@x`.
+- The query contains the functions that cannot be cached: `database()`, `current_user`, `current_role`, `user`, `connection_id`, `last_insert_id`, `row_count`, `version`, and `like`.
+- The query contains `?` after `Limit`, such as `Limit ?` and `Limit 10, ?`. Such queries are not cached because the specific value of `?` has a great impact on the queries.
```

Comment:
```suggestion
- The query contains `?` after `Limit`, such as `Limit ?` and `Limit 10, ?`. Such queries are not cached because the specific value of `?` has a great impact on query performance.
```